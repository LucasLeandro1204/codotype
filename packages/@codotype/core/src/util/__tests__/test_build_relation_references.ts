import {
    buildRelationReferences,
    inflateSchemas,
    inflateSchema,
} from "../inflate";
import {
    SchemaInput,
    testState,
    RelationReference,
    InflatedSchema,
} from "../../";

const { userSchema, movieSchema } = testState;

// // // //

const testCases: [string, SchemaInput, SchemaInput[]][] = [
    ["basic schemas", userSchema, [userSchema, movieSchema]],
];

describe("/lib/buildRelationReferences.js", () => {
    testCases.forEach((testCase) => {
        test(testCase[0], () => {
            const expectedReferences: RelationReference[] = buildRelationReferences(
                {
                    schema: testCase[1],
                    schemas: testCase[2],
                },
            );
            expect(expectedReferences).toMatchSnapshot();
        });
    });
});

// // // //

describe("/lib/inflateSchema.js", () => {
    testCases.forEach((testCase) => {
        test(testCase[0], () => {
            const expectedSchema: InflatedSchema = inflateSchema({
                schema: userSchema,
                schemas: [userSchema, movieSchema],
            });
            expect(expectedSchema).toMatchSnapshot();
        });
    });
});

// // // //

describe("/lib/inflateSchemas.js", () => {
    testCases.forEach((testCase) => {
        test(testCase[0], () => {
            const expectedSchemas: InflatedSchema[] = inflateSchemas({
                schemas: [userSchema, movieSchema],
            });
            expect(expectedSchemas).toMatchSnapshot();
        });
    });
});
