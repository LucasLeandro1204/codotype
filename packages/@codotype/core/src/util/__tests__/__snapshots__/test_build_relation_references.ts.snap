// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/lib/buildRelationReferences.js basic schemas 1`] = `
Array [
  Object {
    "destinationSchemaId": "12345",
    "id": "c9a9d172-c2af-47ca-baa1-aa8516d6bf90",
    "identifiers": Object {
      "destination": Object {
        "alias": Object {
          "plural": Object {
            "camel": "directors",
            "kebab": "directors",
            "label": "Directors",
            "pascal": "Directors",
            "snake": "directors",
          },
          "singular": Object {
            "camel": "director",
            "kebab": "director",
            "label": "Director",
            "pascal": "Director",
            "snake": "director",
          },
        },
        "canonical": Object {
          "plural": Object {
            "camel": "users",
            "kebab": "users",
            "label": "Users",
            "pascal": "Users",
            "snake": "users",
          },
          "singular": Object {
            "camel": "user",
            "kebab": "user",
            "label": "User",
            "pascal": "User",
            "snake": "user",
          },
        },
      },
      "source": Object {
        "alias": Object {
          "plural": Object {
            "camel": "directedMovies",
            "kebab": "directed-movies",
            "label": "Directed Movies",
            "pascal": "DirectedMovies",
            "snake": "directed_movies",
          },
          "singular": Object {
            "camel": "directedMovie",
            "kebab": "directed-movie",
            "label": "Directed Movie",
            "pascal": "DirectedMovie",
            "snake": "directed_movie",
          },
        },
        "canonical": Object {
          "plural": Object {
            "camel": "movies",
            "kebab": "movies",
            "label": "Movies",
            "pascal": "Movies",
            "snake": "movies",
          },
          "singular": Object {
            "camel": "movie",
            "kebab": "movie",
            "label": "Movie",
            "pascal": "Movie",
            "snake": "movie",
          },
        },
      },
    },
    "sourceRelationId": "relation-example-01",
    "sourceSchemaId": "45678",
    "type": "BELONGS_TO",
  },
]
`;

exports[`/lib/inflateSchema.js basic schemas 1`] = `
Object {
  "attributes": Array [],
  "configuration": Object {},
  "id": "12345",
  "identifiers": Object {
    "plural": Object {
      "camel": "users",
      "kebab": "users",
      "label": "Users",
      "pascal": "Users",
      "snake": "users",
    },
    "singular": Object {
      "camel": "user",
      "kebab": "user",
      "label": "User",
      "pascal": "User",
      "snake": "user",
    },
  },
  "references": Array [
    Object {
      "destinationSchemaId": "12345",
      "id": "7f4c62f0-84f7-4a5c-8628-9a92e2634e37",
      "identifiers": Object {
        "destination": Object {
          "alias": Object {
            "plural": Object {
              "camel": "directors",
              "kebab": "directors",
              "label": "Directors",
              "pascal": "Directors",
              "snake": "directors",
            },
            "singular": Object {
              "camel": "director",
              "kebab": "director",
              "label": "Director",
              "pascal": "Director",
              "snake": "director",
            },
          },
          "canonical": Object {
            "plural": Object {
              "camel": "users",
              "kebab": "users",
              "label": "Users",
              "pascal": "Users",
              "snake": "users",
            },
            "singular": Object {
              "camel": "user",
              "kebab": "user",
              "label": "User",
              "pascal": "User",
              "snake": "user",
            },
          },
        },
        "source": Object {
          "alias": Object {
            "plural": Object {
              "camel": "directedMovies",
              "kebab": "directed-movies",
              "label": "Directed Movies",
              "pascal": "DirectedMovies",
              "snake": "directed_movies",
            },
            "singular": Object {
              "camel": "directedMovie",
              "kebab": "directed-movie",
              "label": "Directed Movie",
              "pascal": "DirectedMovie",
              "snake": "directed_movie",
            },
          },
          "canonical": Object {
            "plural": Object {
              "camel": "movies",
              "kebab": "movies",
              "label": "Movies",
              "pascal": "Movies",
              "snake": "movies",
            },
            "singular": Object {
              "camel": "movie",
              "kebab": "movie",
              "label": "Movie",
              "pascal": "Movie",
              "snake": "movie",
            },
          },
        },
      },
      "sourceRelationId": "relation-example-01",
      "sourceSchemaId": "45678",
      "type": "BELONGS_TO",
    },
  ],
  "relations": Array [],
}
`;

exports[`/lib/inflateSchemas.js basic schemas 1`] = `
Array [
  Object {
    "attributes": Array [],
    "configuration": Object {},
    "id": "12345",
    "identifiers": Object {
      "plural": Object {
        "camel": "users",
        "kebab": "users",
        "label": "Users",
        "pascal": "Users",
        "snake": "users",
      },
      "singular": Object {
        "camel": "user",
        "kebab": "user",
        "label": "User",
        "pascal": "User",
        "snake": "user",
      },
    },
    "references": Array [
      Object {
        "destinationSchemaId": "12345",
        "id": "74e1bc53-f5cb-41c7-88e4-6adf0341f866",
        "identifiers": Object {
          "destination": Object {
            "alias": Object {
              "plural": Object {
                "camel": "directors",
                "kebab": "directors",
                "label": "Directors",
                "pascal": "Directors",
                "snake": "directors",
              },
              "singular": Object {
                "camel": "director",
                "kebab": "director",
                "label": "Director",
                "pascal": "Director",
                "snake": "director",
              },
            },
            "canonical": Object {
              "plural": Object {
                "camel": "users",
                "kebab": "users",
                "label": "Users",
                "pascal": "Users",
                "snake": "users",
              },
              "singular": Object {
                "camel": "user",
                "kebab": "user",
                "label": "User",
                "pascal": "User",
                "snake": "user",
              },
            },
          },
          "source": Object {
            "alias": Object {
              "plural": Object {
                "camel": "directedMovies",
                "kebab": "directed-movies",
                "label": "Directed Movies",
                "pascal": "DirectedMovies",
                "snake": "directed_movies",
              },
              "singular": Object {
                "camel": "directedMovie",
                "kebab": "directed-movie",
                "label": "Directed Movie",
                "pascal": "DirectedMovie",
                "snake": "directed_movie",
              },
            },
            "canonical": Object {
              "plural": Object {
                "camel": "movies",
                "kebab": "movies",
                "label": "Movies",
                "pascal": "Movies",
                "snake": "movies",
              },
              "singular": Object {
                "camel": "movie",
                "kebab": "movie",
                "label": "Movie",
                "pascal": "Movie",
                "snake": "movie",
              },
            },
          },
        },
        "sourceRelationId": "relation-example-01",
        "sourceSchemaId": "45678",
        "type": "BELONGS_TO",
      },
    ],
    "relations": Array [],
  },
  Object {
    "attributes": Array [
      Object {
        "addons": Object {
          "required": true,
        },
        "datatype": "STRING",
        "defaultValue": null,
        "id": "name-attr",
        "identifiers": Object {
          "camel": "name",
          "kebab": "name",
          "label": "Name",
          "pascal": "Name",
          "snake": "name",
        },
        "internalNote": "the name of the user",
        "locked": false,
        "source": "USER",
      },
      Object {
        "addons": Object {
          "required": true,
        },
        "datatype": "STRING",
        "defaultValue": null,
        "id": "email-attr",
        "identifiers": Object {
          "camel": "email",
          "kebab": "email",
          "label": "Email",
          "pascal": "Email",
          "snake": "email",
        },
        "internalNote": "the email of the user",
        "locked": false,
        "source": "USER",
      },
    ],
    "configuration": Object {},
    "id": "45678",
    "identifiers": Object {
      "plural": Object {
        "camel": "movies",
        "kebab": "movies",
        "label": "Movies",
        "pascal": "Movies",
        "snake": "movies",
      },
      "singular": Object {
        "camel": "movie",
        "kebab": "movie",
        "label": "Movie",
        "pascal": "Movie",
        "snake": "movie",
      },
    },
    "references": Array [],
    "relations": Array [
      Object {
        "destinationSchemaId": "12345",
        "id": "93e5132a-2db9-4e64-b710-c9271540cd46",
        "identifiers": Object {
          "destination": Object {
            "alias": Object {
              "plural": Object {
                "camel": "directors",
                "kebab": "directors",
                "label": "Directors",
                "pascal": "Directors",
                "snake": "directors",
              },
              "singular": Object {
                "camel": "director",
                "kebab": "director",
                "label": "Director",
                "pascal": "Director",
                "snake": "director",
              },
            },
            "canonical": Object {
              "plural": Object {
                "camel": "users",
                "kebab": "users",
                "label": "Users",
                "pascal": "Users",
                "snake": "users",
              },
              "singular": Object {
                "camel": "user",
                "kebab": "user",
                "label": "User",
                "pascal": "User",
                "snake": "user",
              },
            },
          },
          "source": Object {
            "alias": Object {
              "plural": Object {
                "camel": "directedMovies",
                "kebab": "directed-movies",
                "label": "Directed Movies",
                "pascal": "DirectedMovies",
                "snake": "directed_movies",
              },
              "singular": Object {
                "camel": "directedMovie",
                "kebab": "directed-movie",
                "label": "Directed Movie",
                "pascal": "DirectedMovie",
                "snake": "directed_movie",
              },
            },
            "canonical": Object {
              "plural": Object {
                "camel": "movies",
                "kebab": "movies",
                "label": "Movies",
                "pascal": "Movies",
                "snake": "movies",
              },
              "singular": Object {
                "camel": "movie",
                "kebab": "movie",
                "label": "Movie",
                "pascal": "Movie",
                "snake": "movie",
              },
            },
          },
        },
        "sourceRelationId": "relation-example-01",
        "sourceSchemaId": "45678",
        "type": "BELONGS_TO",
      },
    ],
  },
]
`;
